draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -385,128 +355,160 @@ plus a string suitable for computing a digest, but such approaches can easily le
+
+The hash digest MUST be taken over the US-ASCII bytes of the base64url-encoded Disclosure. This follows the convention in JWS [@RFC7515] and JWE [@RFC7516]. The bytes of the hash digest MUST then be base64url-encoded.
+
+Note:
+
+ * The input to the hash function is the base64url-encoded Disclosure, not the bytes encoded by the base64url string.
+ * The bytes of the output of the hash function are base64url-encoded, not the bytes making up the (often used) hex representation of the bytes of the hash digest.
+
+For example, the
+SHA-256 hash digest of the Disclosure `WyI2cU1RdlJMNWhhaiIsICJmYW1pbHlfbmFtZSIsICJNw7ZiaXVzIl0` would be
+`uutlBuYeMDyjLLTpf6Jxi7yNkEF35jdyWMn9U7b_RYY`.
+
+#### Decoy Digests {#decoy_digests}
+
+An Issuer MAY add additional hash digests to the SD-JWT that are not associated with any claim.  The purpose of such "decoy" digests is to make it more difficult for an attacker to see the original number of claims contained in the SD-JWT. It is RECOMMENDED to create the decoy digests by hashing over a cryptographically secure random number. The bytes of the hash digest MUST then be base64url-encoded as above. The same digest function as for the Disclosures MUST be used.
+
+For decoy digests, no Disclosure is sent to the Holder, i.e., the Holder will see hash digests that do not correspond to any Disclosure. See (#decoy_digests_privacy) for additional privacy considerations.
+
+To ensure readability and replicability, the examples in this specification do not contain decoy digests unless explicitly stated.
+
+#### Creating an SD-JWT {#creating_sd_jwt}
+
+An SD-JWT is a JWT that MUST be signed using the Issuer's private key.
+
+An SD-JWT MAY contain both selectively disclosable claims and non-selectively disclosable claims, i.e., claims that are always contained in the SD-JWT in plaintext and are always visible to a Verifier.
+
+Claims controlling the validity of the SD-JWT, such as `iss`, `exp`, or `nbf` are usually included in plaintext. End-User claims MAY be included as plaintext as well, e.g., if hiding the particular claims from the Verifier does not make sense in the intended use case. Ultimately, an Issuer decides which claims are selectively disclosable and which are not.
+
+Plaintext claims are included in the SD-JWT just as they would be in any other JWT.
+
+Selectively disclosable claims are omitted from the SD-JWT. Instead, the hash digests of the respective Disclosures and potentially decoy digests are contained as an array in a new claim, `_sd`.
+
+The `_sd` claim MUST be an array of strings, each string being a hash digests of a Disclosure or a decoy digest as described above.
+The array MAY be empty, although it is RECOMMENDED to omit the claim in this case to save space.
```

Comment:
when can an array be empty..?

Reply:
Depending on the implementation of the Issuer, it might be convenient to just add `_sd` everywhere and leave it empty when it turns out that there's nothing to selectively disclose.

Reply:
```suggestion
The array MAY be empty in case the Issuer decided not to selectively disclose any of the claims at that level. However, it is RECOMMENDED to omit `_sd` claim in this case to save space.
```